contracts/contracts/GrantRound.sol

```diff
@@ -0,0 +1,134 @@
+    require(block.timestamp < endTime, "Time has passed to complete this tx");
+    _;
+  }
+
+  modifier activeRound() {
+    require(block.timestamp >= startTime && block.timestamp < endTime, "Donations must be sent during an active round");
+    _;
+  }
+
+  modifier roundEnd() {
+    require(block.timestamp >= endTime, "Method must be called after the active round has ended");
+    _;
+  }
+
+  /**
+   * @notice Instantiates a new grant round
+   * @param _owner Grant round owner that has permission to payout the matching pool
+   * @param _registry Address that contains the grant metadata
+   * @param _donationToken Address of the ERC20 token in which donations are made
+   * @param _startTime Unix timestamp of the start of the round
+   * @param _endTime Unix timestamp of the end of the round
+   * @param _metaPtr URL pointing to the grant round metadata
+   * @param _minContribution Miniumum donation amount that can be made using the given donation token
+   */
+  constructor(
+    address _owner,
+    address _registry,
+    IERC20 _donationToken,
+    uint256 _startTime,
+    uint256 _endTime,
+    string memory _metaPtr,
+    uint256 _minContribution
+  ) {
+    hasPaidOut = false;
+    owner = _owner;
+    registry = _registry;
+    donationToken = _donationToken;
+    startTime = _startTime;
+    endTime = _endTime;
+    metaPtr = _metaPtr;
+    minContribution = _minContribution;
+  }
+
+  /**
+   * @notice Before the round ends this method accepts matching pool funds
+   * @param _amount The amount of donation token that can be sent to the contract for the matching pool
+   */
+  function acceptMatchingPool(uint256 _amount) external beforeRoundEnd {
+    _safeTransferDonationToken(msg.sender, address(this), _amount);
+  }
+
+  /**
+   * @notice During an active round users can use this method to send donation tokens to a specified grant
+   * @param _donationAmount The number of tokens to be sent to grant receipient
+   * @param _grantId The id of the grant in the registry
+   */
+  function donateToGrant(uint256 _donationAmount, uint96 _grantId) external activeRound {
+    require(_donationAmount >= minContribution, "Donation amount must be greater than minimum contribution");
+
+    GrantRegistry gRegistry = GrantRegistry(registry);
+    require(gRegistry.isGrantInRegistry(_grantId), "Grant with given id does not exist in registry provided");
+
+    address payee = gRegistry.getGrantPayee(_grantId);
+    require(payee != address(0), "Payee not set in the grant metadata");
+    donationToken.transfer(payee, _donationAmount);
+
+    emit DonationSent(_grantId, address(donationToken), _donationAmount, payee, msg.sender);
+  }
+
+  /**
+   * @notice When the round ends the owner can send the remaining matching pool funds to a given address
+   * @param _payoutAddress An address to receive the remaining matching pool funds in the contract
+   */
+  function payoutGrant(address _payoutAddress) external roundEnd onlyOwner {
```

Comment:
I'm not super clear on this function.
Once the round has been completed -> for this version we assume the distribution curve is available.

- Is this not saved on the contract ?
- If not -> is the understanding right that if the round had 5 grants -> the owner would have to invoke this 5 times ?
- Would it make more sense to have this a function which takes in all grantId and how much amount needs to be paid to them -> invoke the bulkTransfer method in a BulkTransfer contract and pay it all in one shot ?
- Also shouldn't hasPaidOut=true be set only after all the payouts are done (cause only then the round is actually complete )

====

Another way of going about this is, where we let the grant owners invoke this function and funds would be claimed by them (but this complicates thing as the contract would have to be aware of how much each grants is supposed to get from the matching pool )

cc @mds1 / @corydickson  / @apbendi  / @phutchins

Reply:
Ignore ^ this I wasn't aware we were letting the actual payout happen via another contract. All chill